dataRetriever/resolvers/peerAuthenticationResolver.go

```diff
@@ -142,59 +151,171 @@ func (res *peerAuthenticationResolver) ProcessReceivedMessage(message p2p.Messag
+		return err
+	}
+
+	if lastErr != nil {
+		lastErr = fmt.Errorf("resolveMultipleHashes last error %w from %d encountered errors", lastErr, errorsFound)
+	}
+	return lastErr
+}
+
+// sendPeerAuthsForHashes sends multiple peer authentication messages for specific hashes
+func (res *peerAuthenticationResolver) sendPeerAuthsForHashes(dataBuff [][]byte, hashesBuff []byte, pid core.PeerID) error {
+	if len(dataBuff) > res.maxNumOfPeerAuthenticationInResponse {
+		return res.sendLargeDataBuff(dataBuff, hashesBuff, res.maxNumOfPeerAuthenticationInResponse, pid)
+	}
+
+	return res.sendData(dataBuff, hashesBuff, 0, 0, pid)
+}
+
+// sendLargeDataBuff splits dataBuff into chunks and sends a message for each
+func (res *peerAuthenticationResolver) sendLargeDataBuff(dataBuff [][]byte, reference []byte, chunkSize int, pid core.PeerID) error {
+	maxChunks := res.getMaxChunks(dataBuff)
+	for chunkIndex := 0; chunkIndex < maxChunks; chunkIndex++ {
+		chunk, err := res.extractChunk(dataBuff, chunkIndex, chunkSize, maxChunks)
+		if err != nil {
+			return err
+		}
+		err = res.sendData(chunk, reference, 0, 0, pid)
+		if err != nil {
+			return err
+		}
 	}
+	return nil
+}
 
-	startingIndex := chunkIndex * maxNumOfPeerAuthenticationInResponse
-	endIndex := startingIndex + maxNumOfPeerAuthenticationInResponse
-	if endIndex > len(peerAuthMsgs) {
-		endIndex = len(peerAuthMsgs)
+// getMaxChunks returns the max num of chunks from a buffer
+func (res *peerAuthenticationResolver) getMaxChunks(dataBuff [][]byte) int {
+	maxChunks := len(dataBuff) / res.maxNumOfPeerAuthenticationInResponse
+	if len(dataBuff)%res.maxNumOfPeerAuthenticationInResponse != 0 {
+		maxChunks++
 	}
-	messagesBuff := peerAuthMsgs[startingIndex:endIndex]
-	chunk := batch.NewChunk(uint32(chunkIndex), hash, uint32(maxChunks), messagesBuff...)
-	return res.marshalAndSend(chunk, pid)
+	return maxChunks
 }
 
-func (res *peerAuthenticationResolver) marshalAndSend(message *batch.Batch, pid core.PeerID) error {
-	buffToSend, err := res.marshalizer.Marshal(message)
+// sendData sends a message to a peer
+func (res *peerAuthenticationResolver) sendData(dataSlice [][]byte, reference []byte, chunkIndex int, maxChunks int, pid core.PeerID) error {
+	b := batch.Batch{
+		Data:       dataSlice,
+		Reference:  reference,
+		ChunkIndex: uint32(chunkIndex),
+		MaxChunks:  uint32(maxChunks),
+	}
+	buffToSend, err := res.marshalizer.Marshal(b)
 	if err != nil {
 		return err
 	}
 
 	return res.Send(buffToSend, pid)
 }
 
-func (res *peerAuthenticationResolver) fetchPeerAuthenticationMessagesForHash(hash []byte) [][]byte {
+// fetchPeerAuthenticationSlicesForHash fetches all peer authentications for the matching pks to hash
+func (res *peerAuthenticationResolver) fetchPeerAuthenticationSlicesForHash(hash []byte) [][]byte {
 	var messages [][]byte
 
 	keys := res.peerAuthenticationPool.Keys()
-	sort.Slice(keys, func(i, j int) bool {
-		return bytes.Compare(keys[i], keys[j]) < 0
-	})
-
 	for _, key := range keys {
```

Comment:
didn't realized how this would have actually end up implemented. 
This is way too sub-optimal to be left as it is.
We should use a get operation here and send the full public key in the request until we figure out how to optimally resolve this.